File: api/src/euro_chess_studio/calculations/pages.py
"""Static definition of the five workshop pages.

Mirrors web/src/lib/pages.ts. Duplicated deliberately: this is five rows of
static data, not worth sharing a codegen pipeline over for a v0.
"""

from dataclasses import dataclass

Modality = str  # one of "meta", "text", "image", "audio", "video"


@dataclass(frozen=True)
class PageDef:
    slug: str
    title: str
    modality: Modality
    order_index: int


PAGES: list[PageDef] = [
    PageDef("presentation", "Presentation", "meta", 0),
    PageDef("chess-machine", "Building a Chess Machine", "text", 1),
    PageDef("painting-pieces", "Painting Our Pieces", "image", 2),
    PageDef("board-sound", "Giving the Board Sound", "audio", 3),
    PageDef("real-world-video", "Video of the Real-World Use Case", "video", 4),
]

File: web/src/lib/pages.ts
/** Static definition of the five workshop pages. Shared shape with api/src/euro_chess_studio/calculations/pages.py. */

export type Modality = "meta" | "text" | "image" | "audio" | "video";

export interface PageDef {
  slug: string;
  title: string;
  modality: Modality;
  order: number;
  showInTabs?: boolean;
}

export const PAGES: PageDef[] = [
  {
    slug: "presentation",
    title: "Presentation",
    modality: "meta",
    order: 0,
    showInTabs: false,
  },
  { slug: "chess-machine", title: "Building a Chess Machine", modality: "text", order: 1 },
  { slug: "painting-pieces", title: "Painting Our Pieces", modality: "image", order: 2 },
  { slug: "board-sound", title: "Giving the Board Sound", modality: "audio", order: 3 },
  {
    slug: "real-world-video",
    title: "Video of the Real-World Use Case",
    modality: "video",
    order: 4,
  },
];

// Keep the presentation page and embedded deck as an emergency fallback,
// without putting a redundant destination in the attendee-facing tabs.
export const PAGE_TABS = PAGES.filter((page) => page.showInTabs !== false);
